source/game/system/CourseMap.cc

```diff
@@ -182,6 +182,434 @@ f32 CourseMap::startTmp3() const {
     return m_startTmp3;
 }
 
+/// @addr{0x80511E00}
+void CourseMap::clearSectorChecked() {
+    for (size_t i = 0; i < m_checkPoint->size(); i++) {
+        getCheckPoint(i)->clearSearched();
+    }
+}
+
+/// @addr{0x80511500}
+s16 CourseMap::findSector(const EGG::Vector3f &pos, u16 checkpointIdx, f32 *checkpointCompletion,
+        bool isRemote) {
+    clearSectorChecked();
+    MapdataCheckPoint *checkpoint = getCheckPoint(checkpointIdx);
+    s16 id = -1;
+    // check if player is in starting checkpoint
+    MapdataCheckPoint::SectorOccupancy completion =
+            checkpoint->checkSectorAndCheckpointCompletion(pos, checkpointCompletion);
+    // flag starting checkpoint as searched
+    checkpoint->setSearched();
+    u32 params = 0;
+    if (isRemote) {
+        params = 6;
+    }
+
+    // LOCAL SEARCH
+    // Searches through checkpoints recursively with a depth limit of 6 (or 12 for online players).
+    // Search stops if the player is found, the depth limit is reached, or a key checkpoint is
+    // encountered.
+    switch (completion) {
+    // Case 1: the player is fully inside the current checkpoint, so just set to current checkpoint
+    case MapdataCheckPoint::SectorOccupancy::InsideSector:
+        id = checkpoint->id();
+        break;
+
+    // Case 2: the player is between the sides of the quad, but NOT between this checkpoint and
+    // next; player is likely in the same checkpoint group
+    case MapdataCheckPoint::SectorOccupancy::OutsideSector_BetweenSides:
```

Comment:
This case statement is 130 lines long. For readability, we should split each of these cases off into helper functions. Otherwise it makes it really hard to read this function in its entirety. For this case, for example, you could call it `findSectorBetweenSides`

Reply:
I agree.